report/sections/Chapter 1/Intro.tex

```diff
@@ -1 +1,53 @@
-\chapter{P7 Intro}
+\chapter{Introduction} \label{chap:introduction}
+On the 1st semester of the Software Master’s Degree at \aau{} the programming paradigms course currently has a problem where most of the attending students failed the course last time it was held. This led to a large survey being sent out and a student task force being assembled to grasp the main flaws with how the course was structured. Here a large part of the students mentioned a lack of expertise in the programming languages taught in the course and a lack of possibilities for actively training their skills during the semester. 
+
+
+With this knowledge in mind, \aau{} chose to restructure the course with a larger focus on practical programming as opposed to the more classic lectures that are usually used at universities. To support this way of teaching, the course lectures have been moved to a more classical classroom setup but with a monitor at each table and a main monitor at the lecture desk. This is meant to allow the students to more easily work together in groups while working on the shared monitor and allowing the lectures to view and share the groups monitors allowing actual implementations by the students to be used for classroom discussions. 
+
+
+This way of solving the problem brings its problems with it, however. As all students are using their own local environments for solving the exercises at hand, other students cannot revisit the exercise solutions discussed as these are not shared, making it hard to revisit them later. Furthermore, it can be hard to remember concrete implementations as well as verify whether your own implementation is correct and lives up to the requirements of the exercise. In fact, it is not uncommon for people to accidentally misunderstand an exercise and therefore implement the wrong solution.  
```

Comment:
"its problem" It is unclear what 'its' refer to.  "problem brings its problems".

What are environments?